src/pykeen/losses.py

```diff
@@ -510,6 +530,199 @@ def forward(
+        positive_negative_balance: float = 0.5,
+        margin_activation: Hint[nn.Module] = 'relu',
+        reduction: str = 'mean',
+    ):
+        r"""Initialize the double margin loss.
+
+        :param positive_margin:
+            The (absolute) margin for the positive scores. Should be larger than the negative one.
+        :param negative_margin:
+            The (absolute) margin for the negative scores. Should be smaller than the positive one.
+        :param offset:
+            TODO @mberr
+        :param positive_negative_balance:
+            The balance between positive and negative term. Must be in (0, 1).
+        :param margin_activation:
+            A margin activation. Defaults to ``'relu'``, i.e. $h(\Delta) = max(0, \Delta + \lambda)$, which is the
+            default "margin loss". Using ``'softplus'`` leads to a "soft-margin" formulation as discussed in
+            https://arxiv.org/abs/1703.07737.
+        :param reduction:
+            The name of the reduction operation to aggregate the individual loss values from a batch to a scalar loss
+            value. From {'mean', 'sum'}.
+        :raises ValueError: If the positive/negative balance is not within the right range
+        """
+        super().__init__(reduction=reduction)
+        if not (0 <= positive_negative_balance <= 1):
+            raise ValueError(
+                f"The positive-negative balance weight must be in (0, 1), but is {positive_negative_balance}",
+            )
+        self.positive_margin, self.negative_margin = self.resolve_margin(
+            positive_margin=positive_margin,
+            negative_margin=negative_margin,
+            offset=offset,
+        )
+        self.negative_weight = 1.0 - positive_negative_balance
+        self.positive_weight = positive_negative_balance
+        self.margin_activation = margin_activation_resolver.make(margin_activation)
+
+    def process_slcwa_scores(
+        self,
+        positive_scores: torch.FloatTensor,
+        negative_scores: torch.FloatTensor,
+        label_smoothing: Optional[float] = None,
+        batch_filter: Optional[torch.BoolTensor] = None,
+        num_entities: Optional[int] = None,
+    ) -> torch.FloatTensor:  # noqa: D102
+        # Sanity check
+        if label_smoothing:
+            raise UnsupportedLabelSmoothingError(self)
+
+        # positive term
+        if batch_filter is None:
+            # implicitly repeat positive scores
+            loss = self.margin_activation(self.positive_margin - positive_scores) * negative_scores.shape[1]
+        else:
+            # negative_scores have already been filtered in the sampler!
+            num_neg_per_pos = batch_filter.shape[1]
+            positive_scores = positive_scores.unsqueeze(dim=1).repeat(1, num_neg_per_pos, 1)[batch_filter]
+            # shape: (nnz,)
+            loss = self.margin_activation(self.positive_margin - positive_scores)
+        loss = self.positive_weight * self._reduction_method(loss)
```

Comment:
this variable nomenclature is a bit confusing, maybe save this as "positive_term" and the other as "negative_term". Or, if you want to save on variable assignment, just do the addition all inline

Reply:
It was a hotfix for line-too-long 😄 

I can change this though